lib/devices/ios/ios-log.js

```diff
@@ -44,91 +40,112 @@ IosLog.prototype.startCapture = function (cb) {
   this.onIosLogStart = cb;
   // Select cmd for log capture
   if (this.udid) {
-    var spawnEnv = _.clone(process.env);
-    var limdDir = path.resolve(__dirname,
-                               "../../../build/deviceconsole");
-    spawnEnv.PATH = process.env.PATH + ":" + limdDir;
-    spawnEnv.DYLD_LIBRARY_PATH = limdDir + ":" + process.env.DYLD_LIBRARY_PATH;
-    logger.debug("Starting iOS device log capture via deviceconsole");
-    // deviceconsole retrieves many old device log lines that came before it was
-    // started so filter those out until we encounter new log lines.
     this.loggingModeOn = false;
-    this.proc = spawn("deviceconsole", ["-u", this.udid], {env: spawnEnv});
-    this.finishStartingLogCapture(cb);
-  } else {
-    var ver = parseInt(this.xcodeVersion.split(".")[0], 10);
-    if (ver >= 5) {
-      var logsPath;
-      if (ver >= 6) {
-        logger.debug("Starting iOS 8.* simulator log capture");
-        if (_.isUndefined(this.simUdid)) {
-          return cb(new Error("iOS8 log capture requires a sim udid"));
-        }
-        logsPath = path.resolve(process.env.HOME, "Library", "Logs",
-                                "CoreSimulator", this.simUdid);
+    var spawnEnv = _.clone(process.env);
+
+    logger.debug("Attempting iOS device log capture via libimobiledevice idevicesyslog");
+    which("idevicesyslog", function (err) {
+      if (!err) {
+        this.proc = spawn("idevicesyslog", {env: spawnEnv});
       } else {
-        logger.debug("Starting iOS 7.* simulator log capture");
-        logsPath = path.resolve(process.env.HOME, "Library", "Logs",
-                                "iOS Simulator", "7.*");
+        logger.warn("Could not capture device log using libimobiledevice idevicesyslog. Libimobiledevice probably isn't installed");
+        logger.debug("Attempting iOS device log capture via deviceconsole");
+        var limdDir = path.resolve(__dirname,
+                                   "../../../build/deviceconsole");
+        spawnEnv.PATH = process.env.PATH + ":" + limdDir;
+        spawnEnv.DYLD_LIBRARY_PATH = limdDir + ":" + process.env.DYLD_LIBRARY_PATH;
+
+        // deviceconsole retrieves many old device log lines that came before it was
+        // started so filter those out until we encounter new log lines.
+        this.proc = spawn("deviceconsole", ["-u", this.udid], {env: spawnEnv});
       }
-      var errOnlyCb = function (cb) { return function (err) { cb(err); }; }; // makes waterfall safer
-      var touchLog = function (cb) {
-        async.series([
-          function (cb) { mkdirp(logsPath, errOnlyCb(cb)); },
-          function (cb) { touch(path.resolve(logsPath, "system.log"), errOnlyCb(cb)); },
-          function (cb) {
-            require('fs').appendFile(
-              path.resolve(logsPath, "system.log"),
-              'A new simulator is about to start!',
-              errOnlyCb(cb));
+
+      this.finishStartingLogCapture(cb);
+    }.bind(this));
+
+  } else {
```

Comment:
this function is getting so long now that instead of having this massive conditional, I'd probably split it into two methods, `startCaptureOnRealDevice` and `startCaptureOnSim`